### PR TITLE
feat(exception): EmployeeExceptionHandler was added

### PR DESCRIPTION
**Pull request by Larisa-sh**

### CODE REVIEW

File: src/main/java/com/example/springbootapp/service/EmployeeService.java [hunk 1]
```diff
@@ -24,23 +26,30 @@ public EmployeeService(EmployeeRepository employeeRepository, EmployeeMapper emp
...
+    public List<EmployeeDTO> getAllEmployees() {
 
         return employeeRepository.findAll().stream().map(employeeMapper::employeeToEmployeeDTO).collect(Collectors.toList());
     }
 
-    public EmployeeDTO showEmployeeById(long id) {
+    public EmployeeDTO getEmployeeById(long id) {
         Optional<Employee> foundEmployee = employeeRepository.findById(id);
-        return employeeMapper.employeeToEmployeeDTO(foundEmployee.get());
+        return employeeMapper.employeeToEmployeeDTO(foundEmployee.orElseThrow(()-> new EmployeeNotFoundException("Employee with id " + id + " doesn't exist")));
```
Comment (DzianisYermalovich) [inline src/main/java/com/example/springbootapp/service/EmployeeService.java]: Could we check optional when we try to create Employee local var? As in  updateEmployee method

File: src/main/java/com/example/springbootapp/service/EmployeeService.java [hunk 2]
```diff
@@ -24,23 +26,30 @@ public EmployeeService(EmployeeRepository employeeRepository, EmployeeMapper emp
...
+            throw new EmployeeNotSavedException("Employee not saved because at least one field is null");
+        }
         return employeeRepository.save(employeeMapper.employeeDTOtoEmployee(employeeDTO));
     }
 
     @Transactional
     public Employee updateEmployee(long employeeId, EmployeeDTO updatedEmployeeDTO) {
-        Employee employee = employeeRepository.findById(employeeId).get();
+        Employee employee = employeeRepository.findById(employeeId).orElseThrow(()-> new EmployeeNotFoundException("Employee with id " + employeeId + " doesn't exist"));
+        if (updatedEmployeeDTO.isAtLeastOneParamNull(updatedEmployeeDTO)){
```
Comment (DzianisYermalovich) [inline src/main/java/com/example/springbootapp/service/EmployeeService.java:49]: Could you do this validation before calling DB? It reduces the amount of unnecessary requests into this DB.

**Comment by DzianisYermalovich** [comment]
Add custom handling for all exceptions. Even in case of unexpected error application should support the common format of error response.

**Comment by DzianisYermalovich** [comment]
Add custom handling for all exceptions. Even in case of unexpected error application should support the common format of error response.

**Comment by DzianisYermalovich** [comment]
Add custom handling for all exceptions. Even in case of unexpected error application should support the common format of error response.

**Comment by DzianisYermalovich** [comment]
Add custom handling for all exceptions. Even in case of unexpected error application should support the common format of error response.

**Comment by DzianisYermalovich** [comment]
Add custom handling for all exceptions. Even in case of unexpected error application should support the common format of error response.

**Comment by DzianisYermalovich** [comment]
Add custom handling for all exceptions. Even in case of unexpected error application should support the common format of error response.

**Comment by DzianisYermalovich** [comment]
Add custom handling for all exceptions. Even in case of unexpected error application should support the common format of error response.

**Comment by DzianisYermalovich** [comment]
Add custom handling for all exceptions. Even in case of unexpected error application should support the common format of error response.

File: src/main/java/com/example/springbootapp/service/EmployeeService.java (hunk 1 above)
Comment (Larisa-sh) [reply src/main/java/com/example/springbootapp/service/EmployeeService.java]: method was changed to be similar as updateEmployee method

File: src/main/java/com/example/springbootapp/service/EmployeeService.java (hunk 2 above)
Comment (Larisa-sh) [reply src/main/java/com/example/springbootapp/service/EmployeeService.java:49]: I changed commands positions and now there is validation before calling DB

File: src/main/java/com/example/springbootapp/exception/EmployeeExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class EmployeeExceptionHandler {
+
+    @ExceptionHandler
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
```
Comment (DzianisYermalovich) [inline src/main/java/com/example/springbootapp/exception/EmployeeExceptionHandler.java]: please order all methods from specific exception to common

**Review by DzianisYermalovich — APPROVED** [review]

**Pull request merged**